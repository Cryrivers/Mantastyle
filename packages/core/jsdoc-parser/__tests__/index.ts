import jsdocParser from '../src/index';

describe('jsdocParser', () => {
  it('should parse jsdoc', () => {
    expect(
      jsdocParser(`
    @mantastyle {{ asdf asdf asdf }}
    @random foo "bar"
    @test 123 123 123
    `),
    ).toMatchSnapshot();

    expect(
      jsdocParser(`
    random message here too
    @multiline {{
      asdf 
      "asdf"
      1234
      "1234"
      (manta "rocks!" )
    }}
    random comment around
    `),
    ).toMatchSnapshot();

    expect(
      jsdocParser(`
    random message here too
    @multiline {{
      asdf 
      "asdf"
      1234
      "1234"
      (manta "rocks!" )
    }}

    @example qwer 
    Lorem Ipsum is simply dummy text of the printing and
     typesetting industry. Lorem Ipsum has been the industry
     's standard dummy text ever since the 1500s, 
     when an unknown printer took a galley of type
      and scrambled it to make a type specimen book. 
      It has survived not only five centuries, 


    `),
    ).toMatchSnapshot();
  });

  it('should work with empty strings too', () => {
    expect(jsdocParser('')).toMatchSnapshot();
    expect(
      jsdocParser(`
    
    
    
    `),
    ).toMatchSnapshot();
  });
});
