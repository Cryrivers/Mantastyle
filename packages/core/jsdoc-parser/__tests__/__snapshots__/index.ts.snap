// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jsdocParser should parse jsdoc 1`] = `
Array [
  Object {
    "key": "@mantastyle",
    "value": "{{ asdf asdf asdf }}",
  },
  Object {
    "key": "@random",
    "value": "foo \\"bar\\"",
  },
  Object {
    "key": "@test",
    "value": "123 123 123",
  },
]
`;

exports[`jsdocParser should parse jsdoc 2`] = `
Array [
  Object {
    "key": "@multiline",
    "value": "{{
      asdf 
      \\"asdf\\"
      1234
      \\"1234\\"
      (manta \\"rocks!\\" )
    }}
    random comment around",
  },
]
`;

exports[`jsdocParser should parse jsdoc 3`] = `
Array [
  Object {
    "key": "@multiline",
    "value": "{{
      asdf 
      \\"asdf\\"
      1234
      \\"1234\\"
      (manta \\"rocks!\\" )
    }}",
  },
  Object {
    "key": "@example",
    "value": "qwer 
    Lorem Ipsum is simply dummy text of the printing and
     typesetting industry. Lorem Ipsum has been the industry
     's standard dummy text ever since the 1500s, 
     when an unknown printer took a galley of type
      and scrambled it to make a type specimen book. 
      It has survived not only five centuries,",
  },
]
`;

exports[`jsdocParser should work with empty strings too 1`] = `Array []`;

exports[`jsdocParser should work with empty strings too 2`] = `Array []`;
